pydantic/main.py

```diff
@@ -834,6 +834,10 @@ def _iter(
         exclude_none: bool = False,
     ) -> 'TupleGenerator':
 
+        # merge field set excludes with explicit exclude parameter with explicit overriding field set options.
+        field_exclude = {k: v.field_info.exclude for k, v in self.__fields__.items() if v.field_info.exclude}
```

Comment:
Can we calculate `field_exclude` value once at model creation to save time here?

Reply:
Agreed. Currently, I was just using this as a quick way to test the merging functionality.
I still have to combine this with the "model config" option, and when I do that I'll try to pre-compute this is well.

Reply:
I haven't done this yet, I'm also not sure how much time this saves us given all the other stuff that is done   in the __iter__ function. If everything else looks good and everyone like the design I'm open to storing this on the class somewhere (but we'd need to think about where)

Reply:
> I'm also not sure how much time this saves us given all the other stuff that is done in the iter function

In case when there are no excluded or inclided fields, `._iter()` just yields from `.__dict__.items()`, doing something on top of that will be much slower, I think:
```py
if allowed_keys is None and not (to_dict or by_alias or exclude_unset or exclude_defaults or exclude_none):
    # huge boost for plain _iter()
    yield from self.__dict__.items()
```

Also, maybe it's worth precompute some of the other stuff too: for example compute `allowed_keys` from config at model creation might give some performance boost too.

> I'm open to storing this on the class somewhere (but we'd need to think about where)

Computing it in `ModelMetaClass.__new__` after all fields are resolved and storing in one of the `__slots__` should do the trick?  Or you did you mean something more specific?

Reply:
Did this now, class level exclude/includes are pre-computed and stored on the class.
This does mean that people can't go in and modify the field_info by hand and expect stuff to work... but that's already the case for other attributes as well so I think that's ok.